sklearn/linear_model/logistic.py

```diff
@@ -1788,69 +1873,88 @@ def fit(self, X, y, sample_weight=None):
                       intercept_scaling=self.intercept_scaling,
                       random_state=self.random_state,
                       max_squared_sum=max_squared_sum,
-                      sample_weight=sample_weight
+                      sample_weight=sample_weight,
+                      l1_ratio=l1_ratio
                       )
             for label in iter_encoded_labels
-            for train, test in folds)
-
+            for train, test in folds
+            for l1_ratio in l1_ratios_)
+
+        # _log_reg_scoring_path will output different shapes depending on the
+        # multi_class param, so we need to reshape the outputs accordingly.
+        # After reshaping,
+        # - scores is of shape (n_classes X n_folds X n_Cs . n_l1_ratios)
+        # - coefs_paths is of shape
+        #  (n_classes X n_folds X n_Cs . n_l1_ratios X n_features)
+        # - n_iter is of shape
+        #  (n_classes X n_folds X n_Cs . n_l1_ratios) or
+        #  (1 X n_folds X n_Cs . n_l1_ratios)
+        coefs_paths, Cs, scores, n_iter_ = zip(*fold_coefs_)
+        self.Cs_ = Cs[0]
         if multi_class == 'multinomial':
-            multi_coefs_paths, Cs, multi_scores, n_iter_ = zip(*fold_coefs_)
-            multi_coefs_paths = np.asarray(multi_coefs_paths)
-            multi_scores = np.asarray(multi_scores)
-
-            # This is just to maintain API similarity between the ovr and
-            # multinomial option.
-            # Coefs_paths in now n_folds X len(Cs) X n_classes X n_features
-            # we need it to be n_classes X len(Cs) X n_folds X n_features
-            # to be similar to "ovr".
-            coefs_paths = np.rollaxis(multi_coefs_paths, 2, 0)
-
-            # Multinomial has a true score across all labels. Hence the
-            # shape is n_folds X len(Cs). We need to repeat this score
-            # across all labels for API similarity.
-            scores = np.tile(multi_scores, (n_classes, 1, 1))
-            self.Cs_ = Cs[0]
-            self.n_iter_ = np.reshape(n_iter_, (1, len(folds),
-                                                len(self.Cs_)))
-
+            coefs_paths = np.reshape(
+                coefs_paths,
+                (len(folds),  len(l1_ratios_) * len(self.Cs_), n_classes, -1)
+            )
+            # equiv to coefs_paths = np.moveaxis(coefs_paths, (0, 1, 2, 3),
+            #                                                 (1, 2, 0, 3))
+            coefs_paths = np.swapaxes(coefs_paths, 0, 1)
```

Comment:
you find two swapaxes easier to read than one rollaxis? I don't actually have an opinion as the comment above makes it pretty clear.

Reply:
I did at the time when I was trying to break it down but not anymore ^^